### ISSUE TITLE
The verifier always returns exit code 0

### ISSUE THREAD

**Issue by bjorn-spire** [pact-foundation/pact-python#83]
When invoking the underlying Ruby library the popen call doesn't communicate and as such always returns the exit code `None` which ends up being 0.

https://github.com/pact-foundation/pact-python/blob/589224a93c9dab26be0b5e0d4e86de64f1b3dff7/pact/verify.py#L139-L143

If the code is changed to issue `communicate` on the pipe it'll wait and return the actual `returncode` at the end, so we can use the exit code properly. :)

That is:

```python
    p = subprocess.Popen(command, bufsize=1, env=env, stdout=subprocess.PIPE,
                         stderr=subprocess.STDOUT, universal_newlines=True)

    sanitize_logs(p, verbose)
    try:
        p.communicate(timeout=timeout * 2)
    except TimeoutExpired:
        p.kill()

    sys.exit(p.returncode)
```

I'm not actually running the verifier from the CLI but rather I'm invoking the `main` from inside a test, because I want pact to just run as part of my whole suite instead of as a completely separate process. It was a bit iffy to make invoke click, but possible. That said, it would be nice if invoking the verifier from inside of Python code didn't have to be done through click and instead on a normal python function. :)

**Comment by cwilliamson226** [comment]
Any word on this issue? Seems pretty serious, having trouble properly integrating this into our CI workflow since we can't detect the error condition.

**Comment by bethesque** [comment]
That is pretty serious. You could always call the pact-provider-verifier directly until this is fixed. https://github.com/pact-foundation/pact-provider-verifier

**Comment by matthewbalvanz-wf** [comment]
I'm on the road this week, but that might help me find some time to look at this issue. Using `communicate` would fix the issue, but that would cause the logs to be held until the process finished instead of printing as they occurred. I think we might be able to use `poll` or `wait` after we've finished printing logs to get the return code to populate properly. I'll try to do some experiments to confirm that.

**Comment by matthewbalvanz-wf** [comment]
This should be resolved with the v0.17.0 release.
